frontend/src/services/tour.service.js

```diff
@@ -43,6 +44,104 @@ class TourService {
     return response.status === 200 ? response.data : [];
   }
 
+  /**
+   * Add a new entry to the tour endpoint.
+   *
+   * The data dict can have the following keys.
+   * - name (string)
+   * - region (url of an entry)
+   * - buildings (list with dict: building (url) and order_index (int))
+   *
+   * @param data dict with the data.
+   * @returns {Promise<*>}
+   */
+  async post(data) {
+    const tourData = { name: data.name, region: data.region };
+    const response = await ApiInstance.getApi().post("tour/", tourData);
+
+    for (const building of data.buildings) {
+      await buildingInTourService.post({
+        tour: response.data.url,
+        order_index: building.order_index,
+        building: building.building,
+      });
+    }
+    return response.data;
+  }
+
+  /**
+   * Update a tour by id. This creates a new tour entry.
+   *
+   * The data dict can have the following keys.
+   * - name (string)
+   * - region (url of an entry)
+   * - buildings (list with dict: building (url) and order_index (int))
+   *
+   * @param id ID of the entry you want to update.
+   * @param data Dict, data you want to chance.
+   * @returns {Promise<*>}
+   */
+  async patchById(id, data) {
+    const old_data = this.getById(id);
+    if (!("name" in data)) {
+      data.name = old_data.name;
+    }
+    if (!("region" in data)) {
+      data.region = old_data.name;
```

Comment:
Should be old_data.region

Reply:
old_data is the previous entry of this tour. I want to check if "name" and "region" is already in the new data, because I do a post after this (all arguments needs to be added if there were not yet present).

Reply:
i understand what you're doing but shouldn't it be data.region = old_data.region? Otherwise you're setting the tour name as the region url

Reply:
This is fixed now!